conformance/lang/expressions/logical-expressions/logical_and.balt

```diff
@@ -0,0 +1,732 @@
+
+class Person1 {
+    boolean a = false;
+
+    function init() {
+        self.a = true;
+    }
+}
+
+function init() {
+    Person1 newPerson = new Person1();
+    io:println(true && newPerson.a); // @output true
+    io:println(newPerson.a && true); // @output true
+    io:println(newPerson.a && newPerson.a); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using type cast expression.
+Labels: logical-expr, type-cast-expr
+
+type A true;
+
+function init() {
+    A a = true;
+    io:println(true && <boolean>a); // @output true
+    io:println(<boolean>a && true); // @output true
+    io:println(false && <boolean>a); // @output false
+    io:println(<boolean>a && false); // @output false
+    io:println(<boolean>a && <boolean>a); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using unary expression.
+Labels: logical-expr, unary-expr
+
+function init() {
+    io:println(true && !false); // @output true
+    io:println(!false&& true); // @output true
+    io:println(!true&& false); // @output false
+    io:println(false&& !true); // @output false
+    io:println(!true && !false); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression using relational expression.
+Labels: logical-expr, relational-expr
+
+function init() {
+    io:println(true && 4>2); // @output true
+    io:println(4>2 && true); // @output true
+    io:println(false && 4>2); // @output false
+    io:println(4>2 && false); // @output false
+    io:println(4>3 && 4>3); // @output true
+
+    io:println(true && 4<2); // @output false
+    io:println(4<2 && true); // @output false
+    io:println(false && 4<2); // @output false
+    io:println(4<2 && false); // @output false
+    io:println(4<3 && 4<3); // @output false
+
+    io:println(true && 4>=2); // @output true
+    io:println(4>=2 && true); // @output true
+    io:println(false && 4>=2); // @output false
+    io:println(4>=2 && false); // @output false
+    io:println(4>=3 && 4>=3); // @output true
+
+    io:println(true && 4<=2); // @output false
+    io:println(4<=2 && true); // @output false
+    io:println(false && 4<=2); // @output false
+    io:println(4<=2 && false); // @output false
+    io:println(4<=3 && 4<=3); // @output false
+
+    io:println(4<2 && 4>2); // @output false
+    io:println(4>=2 && 4<=3); // @output false
+    io:println(4>2 && 4<=2); // @output false
+    io:println(4>=2 && 4<2); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression using is expression.
+Labels: logical-expr, is-expr
+
+function init() {
+    io:println(true && 2 is int); // @output true
+    io:println(2 is int && true); // @output true
+    io:println(false && 2 is int); // @output false
+    io:println(2 is int && false); // @output false
+    io:println(2 is int && 2 is int); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using equality expression.
+Labels: logical-expr, equality-expr
```

Comment:
There is no `equality-expr` label. Please update to use `equality` or `exact-equality` as applicable.